src/sdk/main/src/impl/HttpClient.cc

```diff
@@ -18,101 +18,77 @@
  *
  */
 
-#include "exceptions/CURLException.h"
-
 #include "impl/HttpClient.h"
 
+#include <httplib.h>
+#include <stdexcept>
+#include <string>
+#include <string_view>
+
 namespace Hedera::internal
 {
-HttpClient::HttpClient()
+namespace
 {
-  if (curl_global_init(CURL_GLOBAL_DEFAULT) != 0)
-  {
-    throw CURLException("Failed to initialize libcurl!");
-  }
-}
-
-HttpClient::~HttpClient()
+// The index in a URL to begin searching for the path after the end of the URL scheme ("http://" or "https://").
+const int SCHEME_END_INDEX = 8;
+
+//
+// Perform an HTTP request.
+//
+// @param url    The URL to which to send the request.
+// @param method The HTTP method type of this request.
+// @param body   The body of the request.
+// @return The response of the request.
+//
+[[nodiscard]] std::string performRequest(std::string_view url, std::string_view method, std::string_view body)
 {
-  curl_global_cleanup();
-}
+  // Create an HTTP client to communicate with the given URL.
+  httplib::Client client(std::string(url.substr(0, url.find('/', SCHEME_END_INDEX))));
+  const std::string path = url.substr(url.find('/', SCHEME_END_INDEX)).data();
 
-// example mirrorNode query:
-// httpClient.invokeREST("https://testnet.mirrornode.hedera.com/api/v1/accounts/" + newAccountId ,"GET", "");
-// note: should time out before calling this function because the mirror node is not updated on time if accountID has
-// been created exactly before the call. Works without timeout if the data in the mirror node is there from some seconds
-// beforehand
-std::string HttpClient::invokeREST(const std::string& url,
-                                   const std::string& httpMethod,
-                                   const std::string& requestBody)
-{
-  std::unique_ptr<CURL, decltype(&curl_easy_cleanup)> curl(curl_easy_init(), curl_easy_cleanup);
+  httplib::Result res;
 
-  if (!curl)
+  // Perform the request based on the HTTP method
+  if (method == "GET")
   {
-    throw CURLException("CURL initialization failed!");
+    res = client.Get(path);
   }
-
-  curl_easy_setopt(curl.get(), CURLOPT_URL, url.c_str());
-
-  std::string response;
-
-  curl_easy_setopt(curl.get(), CURLOPT_WRITEFUNCTION, &HttpClient::writeCallback);
-  curl_easy_setopt(curl.get(), CURLOPT_WRITEDATA, &response);
-
-  curl_easy_setopt(curl.get(), CURLOPT_CUSTOMREQUEST, httpMethod.c_str());
-
-  if (httpMethod == "POST")
+  else if (method == "POST")
+  {
+    res = client.Post(path, body.data(), body.size(), "application/json");
+  }
+  else
```

Comment:
nit: Maybe would look better if we have:
```
const std::unordered_set<std::string_view> supportedMethods = { "GET", "POST" };
```
Check against the set and throw if not in the set.

Reply:
I'm not sure if this would be better, unless I'm misunderstanding. We only have two options of what `method` could be: `"GET"` or `"POST"` and `client` needs to do something different based on each one. I can't think of way using a set that would make this more efficient or cleaner/more readable. Maybe we could make the `method` an `enum`? But even with a set, we'd still have to check `if (method == "GET") { ... }` and `else if (method == "POST") { ... }`. Again, unless I'm misunderstanding something

Reply:
Come to think of it it really would be more readable like you said.

Reply:
```
if (supportedMethods.find(method) == supportedMethods.end()) {
  throw std::invalid_argument("Unsupported HTTP method: " + std::string(method));
}
```
Was thinking something like this